src/tileiterator.jl

```diff
@@ -0,0 +1,211 @@
+################################################################################
+##### CoveredRange
+################################################################################
+
+# a range covered by subranges
+struct CoveredRange{R,S} <: AbstractVector{UnitRange{Int}}
+    offsets::R
+    stopping::S
+end
+
+struct FixedLength
+    length::Int
+end
+
+struct LengthAtMost{N <: Union{Int, Nothing}}
+    maxlength::Int
+    maxstop::N
+end
+
+function compute_stop(offset, stopping::FixedLength)
+    return offset+stopping.length
+end
+function compute_stop(offset, stopping::LengthAtMost)
+    return min(offset+stopping.maxlength, stopping.maxstop)
+end
+
+compute_range(offset, stopping)::UnitRange{Int} = (offset+1):compute_stop(offset, stopping)
+
+Base.size(o::CoveredRange,args...) = size(o.offsets, args...)
+Base.@propagate_inbounds function Base.getindex(o::CoveredRange, inds...)
+    offset = o.offsets[inds...]
+    return compute_range(offset, o.stopping)
+end
+
+################################################################################
+##### RoundedRange
+################################################################################
+struct RoundedRange{R} <: AbstractVector{Int}
+    range::R
+end
+
+Base.@propagate_inbounds function Base.getindex(r::RoundedRange, i)
+    rough = r.range[i]
+    return round(Int, rough)
+end
+
+function roundedrange(start; stop, length)
+    inner = LinRange(start, stop, length)
+    return RoundedRange(inner)
+end
+
+Base.size(o::RoundedRange, args...) = size(o.range, args...)
+
+################################################################################
+##### TileIterator
+################################################################################
+struct TileIterator{N,C} <: AbstractArray{NTuple{N, UnitRange{Int}}, N}
+    covers1d::C
+end
+
+function TileIterator(covers1d::NTuple{N, AbstractVector{UnitRange{Int}}}) where {N}
+    C = typeof(covers1d)
+    return TileIterator{N, C}(covers1d)
+end
+
+function TileIterator(axes::Indices{N}, tilesize::Dims{N}) where {N}
+    TileIterator(axes, RelaxLastTile(tilesize))::TileIterator{N}
+end
+
+"""
+    titr = TileIterator(axes::NTuple{N, AbstractUnitRange}, strategy)
```

Comment:
Would you mind also adding the docs for `TileIterator(axes, tilesize)`? It would be good to also have some words on what `strategy` is.

Reply:
done